### PR TITLE
Rework list screen and add pin icon in detail screen

### PR DESCRIPTION
**Pull request by Armetiis**
This PR introduces significant improvements to the ListScreen (aswell as some changes in the ParkingViewModel), focusing on code organization, UI refinement, and overall user experience.

### Major Changes
**ParkingViewModel Refactor:** Reorganized the file structure for improved readability and maintainability, aswell as added documentation to some undocumented functions (NOTE: great part of the line diff goes to this).

##### ListScreen UI Overhaul:
- Aligned action card text and icons to the right/left for better visual balance.
- Removed horizontal paddings and rounded corners from cards for a sleeker look.
- Reintroduced horizontal dividers to improve list item separation.
- Replaced star icon with heart icon for favorites, enhancing intuitive design.
- Added pin icon for pinned parkings to increase visibility of pinned items.

**UPDATE:**  This screenshots are obsolete, refer to the one in my latest comment

image
image

**Note**: It's true that for now, card's height appears way too big and empty. The CARD_HEIGHT constant was introduced to easily tune this aspect

### New Feature: Pin/Unpin in Parking Detail Screen
- Implemented pin icon display in the parking detail screen.
- Users can now pin and unpin parkings directly from the detail view.

### Additional Improvements
- Fixed alignment issues with misaligned stars and text in the list screen.
- Further simplified ListScreen by removing unnecessary UI elements and reducing filter section arguments.
- Added tests for newly implemented features, increasing overall test coverage.

### Coverage
- Increased list screen coverage to 88%
- Total raises to 81%

image

### CODE REVIEW

**Review by sean-prz — COMMENTED** [review]
I think the transition between Pinned Spots to All spots doesn't stand out enough, because everything is the same color/font/size.

When I quickly scan the screenshots, I don't even see that there is a transition to a section "All spots" under the pinned one.

I really feel like a more pronounced divider or a change in color background is needed. 



_Something like this but pretty_

**Comment by Armetiis** [comment]
What do we think?

**Comment by sean-prz** [comment]
This works yes, let's see what the second reviewer think

File: app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt [hunk 1]
```diff
@@ -186,32 +176,86 @@ class ParkingViewModel(
...
+
+  // ================== Image ==================
+  /**
+   * Fetches the image URL from the cloud storage, This function as to be called after retrieving
+   * the path from the Firestore database.
+   *
+   * @param path the path of the image in the cloud storage (stored in Firestore)
+   * @return [StateFlow] containing the URL of the image
+   */
+  fun fetchImageUrl(path: String): StateFlow<String?> {
```
[the remark this answers is not in the document]
Comment (sean-prz) [reply app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt:189]: I just merged this change to main. 
that change was not in the main when this PR went up. We just have to rebase it now and it should be gone

**Review by matttsss — APPROVED** [review]

**Review by sean-prz — APPROVED** [review]
Nice ! 
thanks for taking into account my feedback !

**Pull request merged**